R/tt_pos_and_access.R

```diff
@@ -939,42 +990,49 @@ setMethod("[", c("VTableTree", "numeric", "numeric"),
           function(x, i, j, ..., drop = FALSE) {
     ## have to do it this way because we can't add an argument since we don't
     ## own the generic declaration
-    keep_topleft <- list(...)[["keep_topleft"]] ## returns NULL if not presesnt
+    keep_topleft <- list(...)[["keep_topleft"]]
     keep_titles <- list(...)[["keep_titles"]] %||% FALSE
+    keep_fnotes <- list(...)[["keep_fnotes"]] %||% FALSE
     reindex_refs <- list(...)[["reindex_refs"]] %||% TRUE
-    if(is.null(keep_topleft))
-        keep_topleft <- NA
-
+    
+    # browser()
     nr <- nrow(x)
     nc <- ncol(x)
     i <- .j_to_posj(i, nr)
     j <- .j_to_posj(j, nc)
 
     ##  if(!missing(i) && length(i) < nr) {
-    if(length(i) < nr) { ## already populated by .j_to_posj
-        keep_topleft <- isTRUE(keep_topleft)
+    if(length(i) < nr) {## already populated by .j_to_posj
+        if (is.null(keep_topleft)) keep_topleft <- FALSE
         x <- subset_by_rownum(x, i,
                               keep_topleft = keep_topleft,
-                              keep_titles = keep_titles)
-    } else {
-        keep_topleft <- !identical(FALSE, keep_topleft)
+                              keep_titles = keep_titles,
+                              keep_fnotes = keep_fnotes)
+    } else if (is.null(keep_topleft)) {
+        keep_topleft <- TRUE
     }
+    
+    # browser()
     ##  if(!missing(j) && length(j) < nc)
     if(length(j) < nc)
         x <- subset_cols(x, j,
                          keep_topleft = keep_topleft,
-                         keep_titles = keep_titles)
-
-    if(length(j) == 1L &&
-       length(i) == 1L &&
-       drop) {
+                         keep_titles = keep_titles,
+                         keep_fnotes = keep_fnotes)
+
+    # Dropping everything
+    if (drop) {
+        if (!(length(j) == 1L && 
+               length(i) == 1L)) {
+            stop("When you need to retrieve single values using drop, use single indexes")    
```

Comment:
Before, in cases where subsetting was not only one element, it just did not care about drop=TRUE in the result. I changed that behavior so it is clearer how drop works

Reply:
ref. issue #475

Reply:
The previous behavior was correct; this should not be an error. `drop =  TRUE` is ignored for `data.frame` when "dropping" doesn't make any sense (i.e., the multi-multi case):

```
> mtcars[1,1, drop = TRUE]
[1] 21

> mtcars[1,1, drop = FALSE]
          mpg
Mazda RX4  21

> mtcars[,1, drop = TRUE]
 [1] 21.0 21.0 22.8 21.4 18.7 18.1 14.3 24.4 22.8 19.2 17.8 16.4 17.3 15.2 10.4
[16] 10.4 14.7 32.4 30.4 33.9 21.5 15.5 15.2 13.3 19.2 27.3 26.0 30.4 15.8 19.7
[31] 15.0 21.4

> mtcars[,1, drop = FALSE]
                     mpg
Mazda RX4           21.0
Mazda RX4 Wag       21.0
Datsun 710          22.8
Hornet 4 Drive      21.4
Hornet Sportabout   18.7
Valiant             18.1
Duster 360          14.3
Merc 240D           24.4
Merc 230            22.8
Merc 280            19.2
Merc 280C           17.8
Merc 450SE          16.4
Merc 450SL          17.3
Merc 450SLC         15.2
Cadillac Fleetwood  10.4
Lincoln Continental 10.4
Chrysler Imperial   14.7
Fiat 128            32.4
Honda Civic         30.4
Toyota Corolla      33.9
Toyota Corona       21.5
Dodge Challenger    15.5
AMC Javelin         15.2
Camaro Z28          13.3
Pontiac Firebird    19.2
Fiat X1-9           27.3
Porsche 914-2       26.0
Lotus Europa        30.4
Ford Pantera L      15.8
Ferrari Dino        19.7
Maserati Bora       15.0
Volvo 142E          21.4

> mtcars[1, , drop = TRUE]
$mpg
[1] 21

$cyl
[1] 6

$disp
[1] 160

$hp
[1] 110

$drat
[1] 3.9

$wt
[1] 2.62

$qsec
[1] 16.46

$vs
[1] 0

$am
[1] 1

$gear
[1] 4

$carb
[1] 4

> mtcars[1:2, 1:2, drop = TRUE]
              mpg cyl
Mazda RX4      21   6
Mazda RX4 Wag  21   6
> 
```

`rtables` should do the same thing, not throw an error